packages/react-dom/src/__tests__/ReactDOMEventListener-test.js

```diff
@@ -26,86 +25,99 @@ describe('ReactDOMEventListener', () => {
     // TODO: can we express this test with only public API?
     ReactDOMComponentTree = require('../client/ReactDOMComponentTree');
     ReactDOMEventListener = require('../events/ReactDOMEventListener');
-    ReactTestUtils = require('react-dom/test-utils');
 
     handleTopLevel = jest.fn();
-    ReactDOMEventListener._handleTopLevel = handleTopLevel;
   });
 
   it('should dispatch events from outside React tree', () => {
+    var mock = jest.fn();
+
+    var container = document.createElement('div');
+    var node = ReactDOM.render(<div onMouseEnter={mock} />, container);
     var otherNode = document.createElement('h1');
-    var component = ReactDOM.render(<div />, document.createElement('div'));
-    expect(handleTopLevel.mock.calls.length).toBe(0);
-    ReactDOMEventListener.dispatchEvent('topMouseOut', {
-      type: 'mouseout',
-      fromElement: otherNode,
-      target: otherNode,
-      srcElement: otherNode,
-      toElement: ReactDOM.findDOMNode(component),
-      relatedTarget: ReactDOM.findDOMNode(component),
-      view: window,
-      path: [otherNode, otherNode],
-    });
-    expect(handleTopLevel.mock.calls.length).toBe(1);
+    document.body.appendChild(container);
+    document.body.appendChild(otherNode);
+
+    var nativeEvent = document.createEvent('MouseEvent');
```

Comment:
I saw another PR using `new MouseEvent(type, args)`, could you check if that one's going to be shorter?

Reply:
Yeah it seems